### PR TITLE
Hybrid flow precompiler

### PR DESCRIPTION
**Pull request by george-giloan**
Added precompile support for do_external steps.

### CODE REVIEW

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java [hunk 1]
```diff
@@ -529,23 +535,15 @@ private StepModellingResult compileStep(
...
         String refId = "";
-        Map<String, Object> doRawData;
-        try {
-            //noinspection unchecked
-            doRawData = (Map<String, Object>) stepRawData.get(SlangTextualKeys.DO_KEY);
-        } catch (ClassCastException ex) {
-            doRawData = new HashMap<>();
-        }
-        if (MapUtils.isEmpty(doRawData)) {
-            errors.add(new RuntimeException("Step: \'" + stepName + "\' has no reference information"));
```
Comment (gituser173) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java]: why did you remove this exception from the errors list?

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/TransformersHandler.java [hunk 2]
```diff
@@ -22,6 +22,10 @@
 import java.util.List;
 import java.util.Map;
 
+import static io.cloudslang.lang.compiler.SlangTextualKeys.DO_EXTERNAL_KEY;
+import static io.cloudslang.lang.compiler.SlangTextualKeys.DO_KEY;
```
Comment (gituser173) [inline cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/TransformersHandler.java]: unused

File: cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java (hunk 1 above)
Comment (victorursan) [reply cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilder.java]: Because we added it at the transformers level (DoTransformer line 25)

**Comment by gituser173** [comment]
had two small comments otherwise it looks good 👍

File: cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilderTest.java [hunk 3]
```diff
@@ -241,27 +241,6 @@ public void stepWithNoDoEntranceThrowsException() throws Exception {
     }
 
     @Test
-    public void stepWithEmptyDoEntranceThrowsException() throws Exception {
```
Comment (Bonczidai) [inline cloudslang-compiler/src/test/java/io/cloudslang/lang/compiler/modeller/ExecutableBuilderTest.java]: Why did you remove this test?

File: cloudslang-compiler/src/test/resources/flow_with_external_steps_and_errors.sl [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+namespace: user.ops
+
+imports:
+  ops: user.ops
+
+flow:
+  name: flow_with_external_steps_and_errors
+  workflow:
+    - DoNothing:
+        do_external:
```
Comment (Bonczidai) [inline cloudslang-compiler/src/test/resources/flow_with_external_steps_and_errors.sl]: this is an erronous case, right? or are we supporting such syntax? do we have a test for it?
